File: 1300/E_Romantic_Glasses.py
# import sys
# input = sys.stdin.readline
# t = int(input())
# for _ in range(t):
#     n = int(input())
#     arr = list(map(int,input().split()))
#     seen = set()
#     curr = 0
#     flag = 1
#     found = False
#     for i in range(n):
#         if flag==1:
#             curr += arr[i]
#             flag = 0
#         else:
#             curr -= arr[i]
#             flag= 1
#         if curr == 0 or curr in seen:
#             found = True
#             break
#         seen.add(curr)
#     print("YES" if found else "NO")

for i in range(1,10**9):
    print(i)

File: 1300/E_Romantic_Glasses.rs
use std::io::{self, Read};
use std::collections::HashSet;

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut iter = input.split_whitespace();

    let t: usize = iter.next().unwrap().parse().unwrap();

    let mut outputs = Vec::with_capacity(t);

    for _ in 0..t {
        let n: usize = iter.next().unwrap().parse().unwrap();

        let mut arr = Vec::with_capacity(n);
        for _ in 0..n {
            let x: i64 = iter.next().unwrap().parse().unwrap();
            arr.push(x);
        }

        let mut seen = HashSet::new();
        let mut curr: i64 = 0;
        let mut flag = true;
        let mut found = false;

        for &val in arr.iter() {
            if flag {
                curr += val;
            } else {
                curr -= val;
            }
            flag = !flag; 

            if curr == 0 || seen.contains(&curr) {
                found = true;
                break;
            }
            seen.insert(curr);
        }

        if found {
            outputs.push("YES");
        } else {
            outputs.push("NO");
        }
    }

    println!("{}", outputs.join("\n"));
}
